azurerm/resource_arm_logic_app_trigger_http_request.go

```diff
@@ -0,0 +1,188 @@
+	err := json.Unmarshal([]byte(schemaRaw), &schema)
+	if err != nil {
+		return fmt.Errorf("Error unmarshalling JSON from Schema: %+v", err)
+	}
+
+	inputs := map[string]interface{}{
+		"schema": schema,
+	}
+
+	if v, ok := d.GetOk("method"); ok {
+		inputs["method"] = v.(string)
+	}
+
+	if v, ok := d.GetOk("relative_path"); ok {
+		// TODO:  Property 'method' is required if 'relativePath' is provided.
+		inputs["relativePath"] = v.(string)
+	}
+
+	trigger := map[string]interface{}{
+		"inputs": inputs,
+		"kind":   "Http",
+		"type":   "Request",
+	}
+
+	logicAppId := d.Get("logic_app_id").(string)
+	name := d.Get("name").(string)
+	err = resourceLogicAppTriggerUpdate(d, meta, logicAppId, name, trigger)
+	if err != nil {
+		return err
+	}
+
+	return resourceArmLogicAppTriggerHttpRequestRead(d, meta)
+}
+
+func resourceArmLogicAppTriggerHttpRequestRead(d *schema.ResourceData, meta interface{}) error {
+	id, err := parseAzureResourceID(d.Id())
+	if err != nil {
+		return err
+	}
+
+	resourceGroup := id.ResourceGroup
+	logicAppName := id.Path["workflows"]
+	name := id.Path["triggers"]
+
+	t, app, err := retrieveLogicAppTrigger(meta, resourceGroup, logicAppName, name)
+	if err != nil {
+		return err
+	}
+
+	if t == nil {
+		log.Printf("[DEBUG] Logic App %q (Resource Group %q) does not contain Trigger %q - removing from state", logicAppName, resourceGroup, name)
+		d.SetId("")
+		return nil
+	}
+
+	trigger := *t
+
+	d.Set("name", name)
+	d.Set("logic_app_id", app.ID)
+
+	v := trigger["inputs"]
+	if v == nil {
+		return fmt.Errorf("Error `inputs` was nil for HTTP Trigger %q (Logic App %q / Resource Group %q)", name, logicAppName, resourceGroup)
+	}
+
+	inputs, ok := v.(map[string]interface{})
+	if !ok {
+		return fmt.Errorf("Error parsing `inputs` for HTTP Trigger %q (Logic App %q / Resource Group %q)", name, logicAppName, resourceGroup)
+	}
+
+	if method := inputs["method"]; method != nil {
+		d.Set("method", method.(string))
+	}
+
+	if relativePath := inputs["relativePath"]; relativePath != nil {
+		d.Set("relative_path", relativePath.(string))
+	}
+
+	if schemaRaw := inputs["schema"]; schemaRaw != nil {
+		schema, err := json.Marshal(schemaRaw)
+		if err != nil {
+			return fmt.Errorf("Error serializing the Schema to JSON: %+v", err)
+		}
+
+		d.Set("schema", string(schema))
+	}
+
+	return nil
+}
+
+func resourceArmLogicAppTriggerHttpRequestDelete(d *schema.ResourceData, meta interface{}) error {
+	id, err := parseAzureResourceID(d.Id())
+	if err != nil {
+		return err
+	}
+
+	resourceGroup := id.ResourceGroup
+	logicAppName := id.Path["workflows"]
+	name := id.Path["triggers"]
+
+	err = resourceLogicAppTriggerRemove(d, meta, resourceGroup, logicAppName, name)
+	if err != nil {
+		return fmt.Errorf("Error removing Trigger %q from Logic App %q (Resource Group %q): %+v", name, logicAppName, resourceGroup, err)
+	}
+
+	return nil
+}
+
+func validateLogicAppTriggerHttpRequestRelativePath(v interface{}, k string) (ws []string, errors []error) {
```

Comment:
so that function checks the string itself is a valid regex, rather than testing for a specific regex; whilst I thought we had a validate method for that I couldn't seem to find it (although I may have missed it)?